### PR TITLE
feat: Add truthy check for binary operators and & or #10

### PR DESCRIPTION
**Pull request by cka29**
@kingwill101 can you change the repo settings so that you can be added as reviewer pls 😊

### CODE REVIEW

File: lib/src/evaluator.dart [hunk 1]
```diff
@@ -139,10 +140,10 @@ class Evaluator implements ASTVisitor<dynamic> {
         result = left >= right;
         break;
       case 'and':
-        result = left && right;
+        result = isTruthy(left) && isTruthy(right);
         break;
       case 'or':
-        result = left || right;
+        result = isTruthy(left) || isTruthy(right);
```
Comment (cka29) [inline lib/src/evaluator.dart:146]: I'm still not sure if the evaluator is the best place to check this. Are there alternatives?

Comment (kingwill101) [reply lib/src/evaluator.dart:146]: It should be fine leaving it there. Currently no other way

**Review by kingwill101 — APPROVED** [review]

**Comment by kingwill101** [comment]
> @kingwill101 can you change the repo settings so that you can be added as reviewer pls 😊

not seeing a setting for that

**Comment by cka29** [comment]
@kingwill101 all good, I guess this is only available working in teams then

**Pull request merged**